### PR TITLE
[Hub Generated] Review request for Microsoft.Azure.Search.Service to add version 2017-11-11-preview

### PR DESCRIPTION
**Pull request by bitsforthought**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/app/branch/brjohnstmsft/azure-rest-api-specs/fix-search).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by brjohnstmsft** [comment]
@jianghaolu Contrary to the PR title, this is not a new API version. These changes are meant to address validation errors that we have never attempted to fix before. See individual commits for details. I have already requested exceptions for the remaining validation errors as I believe them to be limitations or bugs in the validator itself.

**Comment by brjohnstmsft** [comment]
FYI @Yahnoosh 

Also FYI @arv100kri @mgottein since this contains Cognitive Search-related changes. See commits for details. A .NET SDK PR will follow once this is merged.

**Comment by brjohnstmsft** [comment]
@jianghaolu Also, please ignore the auto-generated .NET SDK PR. Our SDK includes custom code and other scripts that run during code generation. The auto-generated PRs will probably never succeed for our .NET SDK. Ask @shahabhijeet if you need details.

File: specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/examples/SearchServiceCreateDataSource.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
         "dataSource": {
             "name": "mydocdbdatasource",
             "description": "My Cosmos DB data source.",
-            "type": "documentdb",
```
Comment (arv100kri) [inline specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/examples/SearchServiceCreateDataSource.json]: Thanks!

File: specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/examples/SearchServiceCreateOrUpdateSkillset.json [hunk 2]
```diff
@@ -8,212 +8,272 @@
...
-			}]
-		}
-	},
-	"responses": {
-		"200": {
-			"body": {
-				"name": "demoskillset",
-				"description": "Extract entities, detect language and extract key-phrases",
-				"skills": [{
 					"@odata.type": "#Microsoft.Skills.Text.NamedEntityRecognitionSkill",
```
Comment (arv100kri) [inline specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/examples/SearchServiceCreateOrUpdateSkillset.json]: Do you mind finding and replacing in the examples usage of "NamedEntityRecognitionSkill" to "EntityRecognitionSkill"... I must've forgot to do this as part of my swagger update where I added entity recognition skill

**Comment by arv100kri** [comment]
Regarding this
>The example for "Get Indexer Execution Status" included properties in the
    response that aren't modeled in the Swagger. These properties actually do
    appear on the wire, but they shouldn't and will be removed in a future API
    version. In the meantime, I have removed them from the example.

@mgottein please comment on whether the comment is accurate or not

File: specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/examples/SearchServiceCreateOrUpdateSkillset.json (hunk 2 above)
Comment (arv100kri) [reply specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/examples/SearchServiceCreateOrUpdateSkillset.json]: The files that require this find and replace are
SearchServiceCreateOrUpdateSkillset.json
SearchServiceCreateSkillset.json
SearchserviceGetSkillset.json
SearchServiceListSkillsets.json

File: specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/searchservice.json [hunk 3]
```diff
@@ -4275,7 +4446,7 @@
         }
       },
       "required": [
-          "@odata.type", "description", "context", "inputs", "outputs"
```
Comment (arv100kri) [inline specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/searchservice.json]: So this is a bit weird... description and context are truly optional...

inputs and outputs are technically optional too, but if left as null for any skill (save the `WebApiSkill`) the API will throw a validation error... I am unsure how to model that in swagger...

**Review by arv100kri — APPROVED** [review]
Generally looks good, I had one request (NamedEntity -> Entity in examples) and the question I posed to @mgottein

**Comment by brjohnstmsft** [comment]
> Regarding this
> 
> The example for "Get Indexer Execution Status" included properties in the
> response that aren't modeled in the Swagger. These properties actually do
> appear on the wire, but they shouldn't and will be removed in a future API
> version. In the meantime, I have removed them from the example.
> 
> @mgottein please comment on whether the comment is accurate or not

@arv100kri I am asserting that it is accurate. :) The Get Indexer Status API re-uses a model class from the Indexing API internally (IndexResult). This is an accident of implementation. That type includes "success" and "statusCode" flags which are meaningless for "ItemError" (errors never succeed, and status code is meaningless for at least some indexer error conditions). I assume that is why these properties are not modeled in the Swagger (it was probably Eugene who did this work, and I would have reviewed and approved it back then).

File: specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/searchservice.json (hunk 3 above)
Comment (brjohnstmsft) [reply specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/searchservice.json]: Will fix; Thanks for confirming

File: specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/examples/SearchServiceCreateOrUpdateSkillset.json (hunk 2 above)
Comment (brjohnstmsft) [reply specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/examples/SearchServiceCreateOrUpdateSkillset.json]: Sure

Comment (brjohnstmsft) [reply specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11-preview/examples/SearchServiceCreateOrUpdateSkillset.json]: I attempted this, but it broke Swagger validation, so I rolled it back. It's up to you and your team to update this.

**Comment by brjohnstmsft** [comment]
@jianghaolu Can you review and merge this please?

**Review by jianghaolu — APPROVED** [review]

**Pull request merged**